WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -339,23 +340,17 @@ private void addCommentApproveActionForCommentNotification(NotificationCompat.Bu
         // adding comment approve action
         Intent commentApproveIntent = getCommentActionIntent();
         commentApproveIntent.addCategory(KEY_CATEGORY_COMMENT_MODERATE);
-        commentApproveIntent.putExtra(NotificationsListFragment.NOTE_INSTANT_APPROVE_EXTRA, true);
+        commentApproveIntent.putExtra(NotificationsProcessingService.ARG_ACTION_TYPE, NotificationsProcessingService.ARG_ACTION_APPROVE);
         if (noteId != null) {
-            commentApproveIntent.putExtra(NotificationsListFragment.NOTE_ID_EXTRA, noteId);
+            commentApproveIntent.putExtra(NotificationsProcessingService.ARG_NOTE_ID, noteId);
         }
-        PendingIntent commentApprovePendingIntent = PendingIntent.getActivity(this, 0, commentApproveIntent,
-                PendingIntent.FLAG_CANCEL_CURRENT);
+        PendingIntent commentApprovePendingIntent =  PendingIntent.getService(this, 0, commentApproveIntent, PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
just in case you want to see what an implementation using `IntentService` would look and behave like, here's the branch were I did my tests (it injects a 5 second delay in each operation so we can see how it is like having to wait for each triggered action to get to completion - it's not worth it fwiw).
Branch here: https://github.com/wordpress-mobile/WordPress-Android/tree/issue/4659-notifications-service-intentservice